### PR TITLE
Increase threadstacksize to 1m since glibC on Ubuntu needs a higher …

### PR DESCRIPTION
**Pull request by radhikaj**
…stack

Signed-off-by: radhikaj <radhikaj@microsoft.com>

### CODE REVIEW

File: tests/aspnetcore5/config.json [hunk 1]
```diff
@@ -18,5 +18,6 @@
         "DOTNET_SYSTEM_GLOBALIZATION_INVARIANT=0",
         "DOTNET_CLI_TELEMETRY_OPTOUT=1"
     ],
+    "ThreadStackSize": "16m",
```
Comment (vtikoo) [inline tests/aspnetcore5/config.json:21]: 16M is quite high, considering dotnet runtime can create a lot of threads. Is the minimum size which works?

**Review by vtikoo — COMMENTED** [review]
@radhikaj Are all of these test suites showing stack overflow failures? If not, a smaller value like 1m might be good for those. Increasing the thread stack size can impact the overall memory requirements.

File: tests/aspnetcore5/config.json (hunk 1 above)
Comment (asvrada) [reply tests/aspnetcore5/config.json:21]: I remember trying out different value for aspnet sample here, and 16m is where it won't crash/hang

**Review by asvrada — APPROVED** [review]
Won't block this PR. But is this the first time dotnet explicitlys throw stack overflow error? We might want to document the recommended tread/stack size configuration for dotnet project in the future

**Comment by radhikaj** [comment]
> @radhikaj Are all of these test suites showing stack overflow failures? If not, a smaller value like 1m might be good for those. Increasing the thread stack size can impact the overall memory requirements.

@vtikoo  - 16m is being used in all the tests that are not showing these stack overflow errors. For other tests, even when there are no errors, there are warnings. I can take an action item to investigate lowering the stack size for some of these. I would like to get the pipeline to pass since these failures might be masking others.

**Comment by radhikaj** [comment]
201077 [2022-02-04T09:32:03.184Z] /home/oeadmin/workspace/Mystikos/Standalone-Pipelines/Unit-Tests-Pipeline/build/bin/myst exec-sgx ext2fs --app-config-path config.json \
201078 [2022-02-04T09:32:03.184Z] /runner/bin/Debug/net5.0/runner /tests/unit-tests.runner.1 ""
201079 [2022-02-04T09:32:11.614Z] [33mmytikos: warn: exec.c(1180): myst_exec(): 
201080 [2022-02-04T09:32:11.614Z]     The thread stack size may be too small for the given program interpreter
201081 [2022-02-04T09:32:11.614Z]     (link loader), which could result in stack overflows. Consider changing
201082 [2022-02-04T09:32:11.614Z]     the thread stack size to at least 1048576 bytes, using the --thread-stack-size
201083 [2022-02-04T09:32:11.614Z]     option or the ThreadStackSize configuration setting.
201084 [2022-02-04T09:32:11.614Z]     [interpreter=/lib64/ld-linux-x86-64.so.2]
201085 [2022-02-04T09:32:11.614Z]     [program=/runner/bin/Debug/net5.0/runner]
201086 [2022-02-04T09:32:11.614Z] 
201087 [2022-02-04T09:32:12.881Z] [0mStack overflow.
201088 [2022-02-04T09:32:12.881Z]    at System.NullReferenceException..ctor()
201089 [2022-02-04T09:32:12.881Z]    at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.AddressRegistrationTests+<ThrowsForUnsupportedAddressFromHosting>d__46.MoveNext()
201090 [2022-02-04T09:32:12.881Z]    at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.__Canon ByRef)
201091 [2022-02-04T09:32:12.881Z]    at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]](System.__Canon ByRef)
201092 [2022-02-04T09:32:12.881Z]    at Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.AddressRegistrationTests.ThrowsForUnsupportedAddressFromHosting(System.String)
201093 [2022-02-04T09:32:12.881Z]    at System.RuntimeMethodHandle.InvokeMethod(System.Object, System.Object[], System.Signature, Boolean, Boolean)
201094 [2022-02-04T09:32:12.881Z]    at System.Reflection.RuntimeMethodInfo.Invoke(System.Object, System.Reflection.BindingFlags, System.Reflection.Binder, System.Object[], System.Globalization.CultureInfo)
201095 [2022-02-04T09:32:12.881Z]    at System.Reflection.MethodBase.Invoke(System.Object, System.Object[])
201096 [2022-02-04T09:32:12.881Z]    at runner.Program.RunTestCase(System.String, System.String)
201097 [2022-02-04T09:32:12.881Z]    at runner.Program.Main(System.String[])




> > @radhikaj Are all of these test suites showing stack overflow failures? If not, a smaller value like 1m might be good for those. Increasing the thread stack size can impact the overall memory requirements.
> 
> @vtikoo - 16m is being used in all the tests that are not showing these stack overflow errors. For other tests, even when there are no errors, there are warnings. I can take an action item to investigate lowering the stack size for some of these. I would like to get the pipeline to pass since these failures might be masking others.

Would you prefer I change it only for the test case that failed since the larger memory requirements might cause tests to error out?

**Comment by vtikoo** [comment]
> Would you prefer I change it only for the test case that failed since the larger memory requirements might cause tests to error out?

For the non-failing ones, we can either let it be(the musl default is 128k), or raise it moderately to something like 512k or 1m.

**Review by vtikoo — APPROVED** [review]

**Pull request merged**